libraries/type-length-value/README.md

```diff
@@ -0,0 +1,104 @@
+# Type-Length-Value
+
+Library with utilities for working with Type-Length-Value structures.
+
+## Example usage
+
+This simple examples defines a zero-copy type with its discriminator.
+
+```rust
+use {
+    bytemuck::{Pod, Zeroable},
+    spl_type_length_value::{discriminator::TlvType, state::{TlvState, TlvStateMut}},
+}
+
+#[repr(C)]
+#[derive(Clone, Copy, Debug, Default, PartialEq, Pod, Zeroable)]
+struct MyValue {
+    data: [u8; 32],
+}
+impl TlvType for MyValue {
+    const TYPE: Discriminator = Discriminator::new([1; Discriminator::LENGTH]);
+}
+let account_size = TlvState::get_base_len() + std::mem::size_of::<MyValue>();
```

Comment:
We can definitely add a function that does something like that for `Sized` types, but beware that it doesn't work with types that have a `String`, `Vec`, or slice in them

Reply:
Actually, thinking about this more, it would have to only be done with a `Pod` or `BorshSchema` type, since `size_of` for a Vector will give you the wrong answer, For example, take a look at this:

```
struct MyStruct {
    a: Vec<usize>,
    b: String,
    c: u64,
}
fn main() {
    println!("Hello, size of MyStruct: {}!", std::mem::size_of::<MyStruct>());
}
```
outputs `56`